.buildkite/scripts/generate_filebeat_pipeline.sh

```diff
@@ -0,0 +1,231 @@
+          mage build unitTest
+        notify:
+          - github_commit_status:
+              context: "Filebeat: windows 2022/Unit Tests"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_WIN_2022}"
+          machine_type: "${GCP_WIN_MACHINE_TYPE}"
+          disk_size: 200
+          disk_type: "pd-ssd"
+        artifact_paths: "${BEATS_PROJECT_NAME}/build/*.*"
+
+YAML
+else
+  echo "The conditions don't match to requirements for generating pipeline steps."
+  exit 0
+fi
+
+echo "Check and add the Extended Tests into the pipeline"
+
+if are_conditions_met_arm_tests || are_conditions_met_macos_tests; then
+  cat >> $pipelineName <<- YAML
+
+  - group: "Extended Tests"
+    key: "extended-tests"
+    steps:
+YAML
+fi
+
+if are_conditions_met_macos_tests; then
+  cat >> $pipelineName <<- YAML
+
+      - label: ":mac: MacOS Unit Tests"
+        key: "macos-unit-tests-extended"
+        command: "cd $BEATS_PROJECT_NAME && mage build unitTest"
+        notify:
+          - github_commit_status:
+              context: "Filebeat: MacOS Unit Tests"
+        agents:
+          provider: "orka"
+          imagePrefix: "${IMAGE_MACOS_X86_64}"
+        artifact_paths: "${BEATS_PROJECT_NAME}/build/*.*"
+
+      - label: ":mac: MacOS ARM Unit Tests"
+        key: "macos-arm64-unit-tests-extended"
+        command: "cd $BEATS_PROJECT_NAME && mage build unitTest"
+        notify:
+          - github_commit_status:
+              context: "Filebeat: MacOS ARM Unit Tests"
+        agents:
+          provider: "orka"
+          imagePrefix: "${IMAGE_MACOS_ARM}"
+        artifact_paths: "${BEATS_PROJECT_NAME}/build/*.xml"
+YAML
+fi
+
+if are_conditions_met_arm_tests; then
+  cat >> $pipelineName <<- YAML
+
+      - label: ":linux: ARM Ubuntu Unit Tests"
+        key: "extended-arm64-unit-test"
+        command: "cd $BEATS_PROJECT_NAME && mage build unitTest"
+        notify:
+          - github_commit_status:
+              context: "Filebeat: Unit Tests ARM"
+        agents:
+          provider: "aws"
+          imagePrefix: "${AWS_IMAGE_UBUNTU_ARM_64}"
+          instanceType: "${AWS_ARM_INSTANCE_TYPE}"
+        artifact_paths: "${BEATS_PROJECT_NAME}/build/*.xml"
+YAML
+fi
+
+if are_conditions_met_win_tests; then
+  cat >> $pipelineName <<- YAML
+
+  - group: "Windows Extended Testing"
+    key: "extended-tests-win"
+    steps:
+      - label: ":windows: Win 2019 Unit Tests"
+        key: "windows-extended-2019"
+        command: |
+          Set-Location -Path $BEATS_PROJECT_NAME
+          mage build unitTest
+        notify:
+          - github_commit_status:
+              context: "Filebeat: Win-2019 Unit Tests"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_WIN_2019}"
+          machine_type: "${GCP_WIN_MACHINE_TYPE}"
+          disk_size: 200
+          disk_type: "pd-ssd"
+        artifact_paths: "${BEATS_PROJECT_NAME}/build/*.*"
+
+      - label: ":windows:-11 Unit Tests"
```

Comment:
```suggestion
      - label: ":windows: Windows 11 Unit Tests"
```

The label in the UI feels strange, so adding the prefix IMHO makes it more precise. The same applies to the other Windows steps.

Reply:
fiexed